src/octokit.ts

```diff
@@ -36,54 +58,81 @@ export interface IRepositoryInfo {
   sha: string;
 }
 
+const extractSha = (eventName: string, event: any): string => {
+  switch (eventName) {
+    case 'pull_request':
+      return event.pull_request.head.sha;
+    case 'push':
+      return event.after;
+    default:
+      warning(event);
+      throw new Error(`Unsupported event '${eventName}'`);
```

Comment:
Well, the idea of fp-ts is not to throw exception, so this is not ok. This should return an Either and a Left instance when something goes wrong.